webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -462,6 +464,58 @@ func (v validator) validateNetworkInterfaceSpec(
 	return allErrs
 }
 
+// mtu and routes is available only with CloudInit bootstrap providers.
+// nameservers and searchDomains is available only with the following bootstrap
+// providers: CloudInit, LinuxPrep, and Sysprep (except for RawSysprep).
+func (v validator) validateNetworkSpecWithBootstrap(
+	interfacePath *field.Path,
+	interfaceSpec vmopv1.VirtualMachineNetworkInterfaceSpec,
+	vm *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+	cloudInit := vm.Spec.Bootstrap.CloudInit
+	linuxPrep := vm.Spec.Bootstrap.LinuxPrep
+	sysPrep := vm.Spec.Bootstrap.Sysprep
```

Comment:
Nit: Sysprep should be a single word :)

Reply:
Its purpose is to differentiate from the imported package:	"github.com/vmware-tanzu/vm-operator/api/v1alpha2/sysprep"